src/datamodel/game_manager.rs

```diff
@@ -0,0 +1,31 @@
+use crate::datamodel::game_status::GameStatus;
+use crate::datamodel::player_action_manager::PlayerActionManager;
+use crate::datamodel::turn_manager::TurnManager;
+use crate::datamodel::Game;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+pub struct GameManager<'a> {
+    pub game: Game,
+    pub status: GameStatus<'a>,
+    pub turn_manager: Rc<TurnManager>,
+    pub action_managers: HashMap<String, PlayerActionManager>,
+}
+
+impl GameManager<'_> {
```

Comment:
why do we have a generic lifetimes here?

Reply:
The lifetime in the class comes from the lifetime of the reference. When implementing it, none of the methods require a lifetime, but the class still has a lifetime so I used the generic one. These lifetimes are probably going to disappear, because I think that we probably should be using smart pointers instead of references in the fields of our data structures. However, I propose to make that change in a separate PR, taking it as an opportunity to review the whole design, test the usage of smart pointers and make the whole codebase consistent.